### PR TITLE
feat: find a project by namespace/slug

### PR DESCRIPTION
**Pull request by ciyer**
Add support for finding a unique project given the namespace/slug

/deploy renku-ui=ciyer/3096-merge-urls #persist

### CODE REVIEW

File: components/renku_data_services/project/db.py [hunk 1]
```diff
@@ -163,6 +163,34 @@ async def get_project(self, user: base_models.APIUser, project_id: str) -> model
...
             return project_orm.dump()
 
+    async def get_project_by_namespace_slug(
+        self, user: base_models.APIUser, namespace: str, slug: str
+    ) -> models.Project:
+        """Get one project from the database."""
+        async with self.session_maker() as session:
+            stmt = (
+                select(schemas.ProjectORM)
+                .where(schemas.NamespaceORM.slug == namespace)
```
Comment (mana-alisafaee) [inline components/renku_data_services/project/db.py:173]: Shouldn't we compare against `namespace.lower()` in this statement?

File: components/renku_data_services/project/db.py [hunk 2]
```diff
@@ -163,6 +163,34 @@ async def get_project(self, user: base_models.APIUser, project_id: str) -> model
...
+                .where(schemas.ProjectSlug.namespace_id == schemas.NamespaceORM.id)
+                .where(schemas.ProjectSlug.slug == slug)
```
Comment (mana-alisafaee) [inline components/renku_data_services/project/db.py:175]: Same as above: `slug.lower()`.

File: test/bases/renku_data_services/data_api/test_projects.py [hunk 3]
```diff
@@ -94,6 +94,17 @@ async def test_project_creation(sanic_client, user_headers, regular_user, app_co
         "http://renkulab.io/repository-2",
     }
 
+    # same as above, but using namespace/slug to retreive the pr
+    _, response = await sanic_client.get(
+        f"/api/data/projects/{payload['namespace'].lower()}/{payload['slug']}", headers=user_headers
```
Comment (mana-alisafaee) [inline test/bases/renku_data_services/data_api/test_projects.py:99]: We can remove `.lower()` here if we do the lowercase conversion in the `get_project_by_namespace_slug` function.

**Review by mana-alisafaee — COMMENTED** [review]
Looks good! Please see my comments.

File: components/renku_data_services/project/db.py (hunk 1 above)
Comment (ciyer) [reply components/renku_data_services/project/db.py]: Good point! Fixed in 651ceb960de0d5064cd9c283d7f01d0c1e56c02c

File: components/renku_data_services/project/db.py (hunk 2 above)
Comment (ciyer) [reply components/renku_data_services/project/db.py]: Fixed in 651ceb960de0d5064cd9c283d7f01d0c1e56c02c

File: test/bases/renku_data_services/data_api/test_projects.py (hunk 3 above)
Comment (ciyer) [reply test/bases/renku_data_services/data_api/test_projects.py]: Fixed in 651ceb960de0d5064cd9c283d7f01d0c1e56c02c

**Review by mana-alisafaee — APPROVED** [review]
Thank you!

**Pull request merged**